DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -92,9 +92,12 @@ public async Task<IActionResult> Index(LoginViewModel model)
         }
 
         [HttpGet]
+        [SetDlsSubApplication]
+        [Route("/{dlsSubApplication}/Login/ChooseACentre", Order = 1)]
+        [Route("/Login/ChooseACentre", Order = 2)]
         [ResponseCache(Location = ResponseCacheLocation.None, NoStore = true)]
         [Authorize(Policy = CustomPolicies.BasicUser)]
-        public IActionResult ChooseACentre(string? returnUrl)
+        public IActionResult ChooseACentre(DlsSubApplication dlsSubApplication, string? returnUrl)

```

Comment:
As well as the SetDlsSubApplication attribute, you also need `[TypeFilter(typeof(ValidateAllowedDlsSubApplication))]` on this action now. This prevents e.g. delegates from going to /TrackingSystem/Login/ChooseACentre. (see MyAccountController where we're also doing the same dual route thing where one has a dlsSubApplication)

Reply:
Done